### PR TITLE
Recover https option in hot command

### PR DESCRIPTION
**Pull request by soilSpoon**
```
error: unknown option '--https'
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! working-note@ hot: `mix watch --hot --https`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the working-note@ hot script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I currently have the same problem as above on laravel-mix 6.
Fixes an issue where https is not properly supported when laravel-mix is upgraded to version 6.

### CODE REVIEW

File: src/config.js [hunk 1]
```diff
@@ -22,6 +22,7 @@ module.exports = function () {
          * @type {Object}
          */
         hmrOptions: {
+            isHttps: !!argv.https,
```
Comment (thecrypticace) [inline src/config.js:25]:
```suggestion
            https: !!argv.https,
```

File: src/builder/WebpackConfig.js [hunk 2]
```diff
@@ -93,17 +93,16 @@ class WebpackConfig {
             return;
         }
 
-        let http = process.argv.includes('--https') ? 'https' : 'http';
-        const url = `${http}://${Config.hmrOptions.host}:${Config.hmrOptions.port}/`;
+        const {isHttps, host, port} = Config.hmrOptions;
```
Comment (thecrypticace) [inline src/builder/WebpackConfig.js:96]:
```suggestion
        const {https, host, port} = Config.hmrOptions;
```

File: src/builder/WebpackConfig.js [hunk 3]
```diff
@@ -93,17 +93,16 @@ class WebpackConfig {
...
+        const protocol = isHttps ? 'https' : 'http';
```
Comment (thecrypticace) [inline src/builder/WebpackConfig.js:97]:
```suggestion
        const protocol = https ? 'https' : 'http';
```

File: src/builder/WebpackConfig.js [hunk 4]
```diff
@@ -115,6 +114,8 @@ class WebpackConfig {
 
             public: url,
             liveReload: false,
+            
+            https: isHttps,
```
Comment (thecrypticace) [inline src/builder/WebpackConfig.js:118]:
```suggestion
            https: https,
```

**Comment by thecrypticace** [comment]
Left a few suggestions. Can you also update the types under `types/config.d.ts`?

**Comment by thecrypticace** [comment]
Thanks @soilSpoon, I'll be reviewing this over the weekend.

File: bin/cli.js [hunk 5]
```diff
@@ -24,6 +24,7 @@ async function run() {
         .command('watch')
         .description('Build and watch files for changes.')
         .option('--hot', 'Enable hot reloading.', false)
+        .option('--https', 'Enable https.', false)
         .action(cmd =>
```
Comment (muhammedraqeeb) [inline bin/cli.js:28]: .option( ' --https', 'Enable https.', false)

**Review by muhammedraqeeb — CHANGES_REQUESTED** [review]

**Comment by rleger** [comment]
Can you give a way to make hmr work with https websites ? trying to reproduce code from the mix 5.0 documentation but without success. Thanks !!

https://laravel-mix.com/docs/5.0/hot-module-replacement

**Comment by soilSpoon** [comment]
You can use `npx mix watch --hot`.
but currently, this PR has not been released.
Please try after this PR is released.

**Pull request merged**